src/lib.rs

```diff
@@ -27,10 +28,11 @@ pub struct Inner<T> {
 
 impl<T> Inner<T> {
     fn wakeup_next(&mut self, resource: T) {
-        let mut this = ResourceState::Broken;
+        let mut this = ResourceState::Empty;
```

Comment:
I think that those two lines could be combined somehow. I didn't yet check this, but maybe something like:
`match mem::replace(&mut self.resource, &mut ResourceState::Empty) ... `
See example here:
https://github.com/rust-lang-nursery/futures-rs/blob/ac8b470a15f3c369ad9a600d6a3c3e29ccdda42d/futures-util/src/future/chain.rs#L37

Reply:
@realcr thanks for suggestion! I pushed the change.